File: system-test/sec-tests/javascript/sec_02_connection_string.js
// SEC-02: Hardcoded Connection Strings
// Test: TC-CR-005, TC-CR-008
const mongoose = require('mongoose');
mongoose.connect('mongodb://admin:MyPassw0rd!@prod-db.example.com:27017/mydb');

const { Client } = require('pg');
const client = new Client({
  connectionString: 'postgresql://root:admin123@localhost:5432/testdb'
});

const redis = require('redis');
const redisClient = redis.createClient({ url: 'redis://:secret123@redis.internal:6379' });

File: system-test/sec-tests/java/Sec02ConnectionString.java
// SEC-02: Hardcoded Connection Strings
// Test: TC-CR-005, TC-CR-008
package com.example;

import java.sql.Connection;
import java.sql.DriverManager;

public class DatabaseConfig {
    public Connection getConnection() throws Exception {
        String url = "jdbc:mysql://root:admin123@localhost:3306/testdb";
        return DriverManager.getConnection(url);
    }

    public Connection getPgConnection() throws Exception {
        String url = "jdbc:postgresql://admin:MyPassw0rd!@prod-db.example.com:5432/mydb";
        return DriverManager.getConnection(url);
    }
}
